.github/workflows/semgrep.yml

```diff
@@ -11,6 +11,7 @@ on:
       - canary
       - dev/*
       - "trunk-merge/**"
+  merge_group:
```

Comment:
I don't know if we added `trunk-merge` here by accident, but this does not run on PRs. I added it to the merge queue as well to be consistent, but I'm happy to remove both lines.